django_sirene/management/commands/populate_sirene_database.py

```diff
@@ -1,157 +1,118 @@
+        logger.debug("Downloading %s ...", uri)
+        urlretrieve(uri, filepath)
 
-        :param at: Date on which we want database, default now
-        """
-        listing = lhtml.parse(self.base_url)
-        prev_month = at.replace(day=1) - datetime.timedelta(days=1)
-        last_month_filename = listing.xpath(
-            '//a[contains(text(), "{}{}_L_M")]/@href'.format(
-                prev_month.year,
-                str(prev_month.month).zfill(2)
-            )
-        )[-1]
+    def populate_with_file(self, filename, uri, importer_class, **options):
+        if options["dry"]:
+            print(f"{uri} in {filename}")
+            return
 
-        daily_filenames = []
-        range_num_day = range(
-            int(at.replace(day=1).strftime('%j')) - 1,
-            int(at.strftime('%j')) + 1
-        )
-        for i in range_num_day:
-            filename = listing.xpath('//a[contains(text(), "{}{}_E_Q")]/@href'.format(
-                at.year,
-                str(i).zfill(3)
-            ))
-            if filename:
-                daily_filenames.append(filename[0])
+        filepath = os.path.join(self.local_csv_path, filename)
 
-        return [last_month_filename] + daily_filenames
-
-    def handle(self, *args, **options):
-
-        if options.get('at'):
-            at = datetime.datetime.strptime(options['at'], '%Y-%m-%d')
-            filenames = self._get_filenames(at=at)
+        if not os.path.exists(filepath) or options["force"]:
+            logger.debug("Downloading data to file %s", filepath)
+            self._download_file(uri, filepath)
+            zfile = zipfile.ZipFile(filepath, "r")
         else:
-            filenames = self._get_filenames()
-
-        files_already_parsed = set(
-            list(
-                Institution.objects.values_list(
-                    'updated_from_filename',
-                    flat=True
-                ).distinct()
-            )
-        )
+            logger.debug("Using already created local file %s", filepath)
+            try:
+                zfile = zipfile.ZipFile(filepath, "r")
+            except zipfile.BadZipfile:
+                logger.debug("Failed to open already created local file %s", filepath)
+                self._download_file(filename, filepath)
+                zfile = zipfile.ZipFile(filepath, "r")
+        # zfile = zipfile.ZipFile(filepath, 'r')
 
-        for filename in filenames:
+        csv_filename = zfile.namelist()[0]
+        assert os.path.splitext(csv_filename)[-1].lower() == ".csv"
 
-            if options['dry']:
-                print(filename)
-                continue
+        with zfile.open(csv_filename) as csv_file:
+            self._import_csv(csv_file, importer_class, force="force" in options)
+        zfile.close()
 
-            filepath = os.path.join(self.local_csv_path, filename)
+        logger.debug("Import %s finshed", csv_filename)
```

Comment:
There is an "i" lacking in finshed, but I thought it could be simpler.
```suggestion
        logger.debug("%s imported", csv_filename)
```